src/progress.rs

```diff
@@ -54,12 +55,98 @@ impl Default for ProgressState {
     }
 }
 
-#[derive(Clone, Debug, Default)]
-struct Configuration {
+/// A Raft internal representation of a Configuration.
+///
+/// This is corollary to a ConfState, but optimized for `contains` calls.
+#[derive(Clone, Debug, Default, PartialEq, Getters)]
+pub struct Configuration {
+    /// The voter set.
+    #[get = "pub"]
     voters: FxHashSet<u64>,
+    /// The learner set.
+    #[get = "pub"]
     learners: FxHashSet<u64>,
 }
 
+impl Configuration {
+    /// Create a new configuration with the given configuration.
+    pub fn new(
+        voters: impl IntoIterator<Item = u64>,
+        learners: impl IntoIterator<Item = u64>,
+    ) -> Self {
+        Self {
+            voters: voters.into_iter().collect(),
+            learners: learners.into_iter().collect(),
+        }
+    }
+}
+
+impl<I> From<(I, I)> for Configuration
```

Comment:
Maybe `(Iter1, Iter2)` is better? We don't need they are same type.

Reply:
Thanks, I was lazy it seems. :(